src/hotspot/share/opto/vectornode.hpp

```diff
@@ -853,6 +853,47 @@ class VectorMaskGenNode : public TypeNode {
    const Type* _elemType;
 };
 
+class VectorMaskOpNode : public TypeNode {
+ public:
+  VectorMaskOpNode(Node* mask, const Type* ty, const Type* ety, int mopc):
```

Comment:
`ty`/`ety` caught my eye. It doesn't match anything in vectornode.hpp and may confuse readers.
Any reason not to use `vt`? 

Also, any particular reason to cache full-blown type instead of capturing just the `BasicType`?

Reply:
In this case all the mask operations produce an integer value.  Thus did not use vt, have removed ety since there its does have any direct use currently.